crates/jstz_api/src/file/file.rs

```diff
@@ -0,0 +1,291 @@
+//! `jstz`'s implementation of JavaScript's `File` API.
+//!
+//! More information:
+//!  - [MDN documentation][mdn]
+//!  - [W3C `File` specification][spec]
+//!  - [WHATWG `Infra` specification][infra-spec]
+//!
+//! [mdn]: https://developer.mozilla.org/en-US/docs/Web/API/Blob
+//! [spec]: https://w3c.github.io/FileAPI/
+//! [infra-spec]: https://infra.spec.whatwg.org/
+
+use boa_engine::{
+    js_string,
+    object::{
+        builtins::{JsDate, JsPromise},
+        Object,
+    },
+    property::Attribute,
+    value::TryFromJs,
+    Context, JsArgs, JsError, JsNativeError, JsResult, JsValue, NativeFunction,
+};
+use boa_gc::{Finalize, GcRefMut, Trace};
+use jstz_core::{
+    accessor,
+    native::{register_global_class, Accessor, JsNativeObject, NativeClass},
+    value::IntoJs,
+};
+
+use super::blob::{Blob, BlobClass, BlobParts, BlobPropertyBag};
+
+#[derive(Trace, Finalize, Clone)]
+pub struct File {
+    blob: Blob,
+    name: String,
+    last_modified: i64,
+}
+
+impl File {
+    // https://w3c.github.io/FileAPI/#file-constructor
+    pub fn new(
+        file_bits: BlobParts,
+        file_name: String,
+        options: &Option<FilePropertyBag>,
+        context: &mut Context<'_>,
+    ) -> JsResult<Self> {
+        // 1. Let bytes be the result of processing blob parts given fileBits and options.
+        //    NOTE: we use Blob constructor instead
+        let blob_property_bag = options
+            .as_ref()
+            .map(|options| options.blob_property_bag.clone());
+        let blob = Blob::new(Some(file_bits), &blob_property_bag, context).unwrap();
```

Comment:
Bubble this error up, don't `.unwrap`. 

General rule: Avoid `.unwrap` or `.expect`. They implicitly add panics. They are only suitable if they have a provided `// SAFETY` comment explaining why it is safe to `.unwrap` or `.expect`.

Reply:
Fixed, thanks!